src/olympia/addons/indexers.py

```diff
@@ -268,6 +268,13 @@ def get_mapping(cls):
                             },
                         },
                     },
+                    'promoted': {
+                        'type': 'object',
+                        'properties': {
+                            'category': {'type': 'text'},
+                            'applications': {'type': 'keyword'}
```

Comment:
we store application compatibility as byte elsewhere so I think we should do the same here.  wdyt?

(The serializer would need some changes though.  I think you'd need to make a new serializer just for the promoted field - the field for applications would be `ReverseChoiceField` inside it)

Reply:
I'll give that a try. I was wondering how to specify that this is a list, and not just a single value. Is that not important to do? Would it just be `'applications': {'type': 'byte'}`?

Reply:
Everything is technically a list (of the specified type) in ES as any field can hold multiple values.

Reply:
Does this need to match what's returned by the API to a caller? The API will be returning `promoted: {category: 'recommended', apps: ['firefox', 'android']}`, but as I understand the suggestion here, this actually wouldn't result in a list, but rather a single number representing what's stored in `promotedaddon.application_id`, so that seems like a mismatch. Is that an issue? I don't fully understand exactly how all this stuff is used.

Reply:
I'm trying to figure out how to implement what @eviljeff suggested and I'm a bit stuck. If the field for `applications` is to be a `ReverseChoiceField`, wouldn't the `Model` need to be `PromotedAddon` and not `Addon`? If we do that, then it seems like we'd have to duplicate some of the logic that currently exists in `Addon.promoted`. I tried just creating a new serializer for `promoted`, as suggested, and including inline code that calculates what I need, but tests are failing because it's causing a query to be run. I'm wondering if a better approach would be to create a new model property on `Addon` which can return the `promoted` dict in the format that ES wants, and then just include that in the ES serializer?